diff -uprN a/mt7603_wifi/common/cmm_info.c b/mt7603_wifi/common/cmm_info.c
--- a/mt7603_wifi/common/cmm_info.c	2019-02-22 10:01:55.000000000 +0100
+++ b/mt7603_wifi/common/cmm_info.c	2019-12-08 21:47:48.553113878 +0100
@@ -132,7 +132,7 @@ INT Set_DriverVersion_Proc(RTMP_ADAPTER
 {
 #ifdef CONFIG_AP_SUPPORT
 	IF_DEV_CONFIG_OPMODE_ON_AP(pAd)
-		DBGPRINT(RT_DEBUG_OFF, ("Driver version-%s %s %s\n", AP_DRIVER_VERSION, __DATE__, __TIME__));
+		DBGPRINT(RT_DEBUG_OFF, ("Driver version-%s\n", AP_DRIVER_VERSION));
 		DBGPRINT(RT_DEBUG_OFF, ("Driver Patch\n"));
 		DBGPRINT(RT_DEBUG_OFF, ("001 : 4AC + MGMT PSE queue pkt stuck detect & AP tr_entry is memory currupted\n"));
 		DBGPRINT(RT_DEBUG_OFF, ("002 : wifi_forward support\n"));
diff -uprN a/mt7603_wifi/os/linux/rt_linux.c b/mt7603_wifi/os/linux/rt_linux.c
--- a/mt7603_wifi/os/linux/rt_linux.c	2019-02-22 10:01:39.000000000 +0100
+++ b/mt7603_wifi/os/linux/rt_linux.c	2019-12-08 22:16:27.246150818 +0100
@@ -47,8 +47,8 @@
 #endif
 
 #if defined(CONFIG_RA_HW_NAT) || defined(CONFIG_RA_HW_NAT_MODULE)
-#include "../../../../../../../../net/nat/hw_nat/ra_nat.h"
-#include "../../../../../../../../net/nat/hw_nat/frame_engine.h"
+#include <../net/nat/hw_nat/ra_nat.h>
+#include <../net/nat/hw_nat/frame_engine.h>
 #endif
 
 /* TODO */
@@ -855,18 +855,27 @@ void RtmpOSFileSeek(RTMP_OS_FD osfd, int
 int RtmpOSFileRead(RTMP_OS_FD osfd, char *pDataPtr, int readLen)
 {
 	/* The object must have a read method */
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3, 19, 0)
 	if (osfd->f_op && osfd->f_op->read) {
 		return osfd->f_op->read(osfd, pDataPtr, readLen, &osfd->f_pos);
-	} else {
-		DBGPRINT(RT_DEBUG_ERROR, ("no file read method\n"));
-		return -1;
+#else
+		if (osfd->f_mode & FMODE_CAN_READ) {
+			return __vfs_read(osfd, pDataPtr, readLen,&osfd->f_pos);
+#endif
+		} else {
+			DBGPRINT(RT_DEBUG_ERROR, ("no file read method\n"));
+			return -1;
+		}
 	}
-}
 
 
 int RtmpOSFileWrite(RTMP_OS_FD osfd, char *pDataPtr, int writeLen)
 {
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4, 1, 0)
 	return osfd->f_op->write(osfd, pDataPtr, (size_t) writeLen, &osfd->f_pos);
+#else
+	return __vfs_write(osfd, pDataPtr, (size_t) writeLen, &osfd->f_pos);
+#endif
 }
 
 
diff -uprN a/mt7603_wifi/os/linux/rt_proc.c b/mt7603_wifi/os/linux/rt_proc.c
--- a/mt7603_wifi/os/linux/rt_proc.c	2019-02-22 10:01:39.000000000 +0100
+++ b/mt7603_wifi/os/linux/rt_proc.c	2019-12-08 21:47:48.553113878 +0100
@@ -61,7 +61,7 @@ int wl_proc_exit(void);
 #endif /* PROCREG_DIR */
 
 #ifdef CONFIG_PROC_FS
-extern struct proc_dir_entry *procRegDir;
+struct proc_dir_entry *procRegDir;
 
 #ifdef VIDEO_TURBINE_SUPPORT
 extern BOOLEAN UpdateFromGlobal;
diff -uprN a/mt7603_wifi/os/linux/rt_profile.c b/mt7603_wifi/os/linux/rt_profile.c
--- a/mt7603_wifi/os/linux/rt_profile.c	2019-02-22 10:01:39.000000000 +0100
+++ b/mt7603_wifi/os/linux/rt_profile.c	2019-12-09 10:27:49.609507632 +0100
@@ -30,8 +30,8 @@
 #endif
 
 #if defined (CONFIG_RA_HW_NAT)  || defined (CONFIG_RA_HW_NAT_MODULE)
-#include "../../../../../../../../net/nat/hw_nat/ra_nat.h"
-#include "../../../../../../../../net/nat/hw_nat/frame_engine.h"
+#include <../net/nat/hw_nat/ra_nat.h>
+#include <../net/nat/hw_nat/frame_engine.h>
 #endif
 
 
@@ -689,7 +689,11 @@ NDIS_STATUS	RTMPReadParametersHook(RTMP_
 		{
 #ifndef OS_ABL_SUPPORT
 			// TODO: need to roll back when convert into OSABL code
-				 fsize = (ULONG)srcf->f_dentry->d_inode->i_size;
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3, 19, 0)
+				fsize = (ULONG)srcf->f_dentry->d_inode->i_size;
+#else
+				fsize = (ULONG)srcf->f_path.dentry->d_inode->i_size;
+#endif
 				if (buf_size < (fsize + 1))
 					buf_size = fsize + 1;
 #endif /* OS_ABL_SUPPORT */
diff -uprN a/mt7603_wifi_ap/Makefile b/mt7603_wifi_ap/Makefile
--- a/mt7603_wifi_ap/Makefile	2019-02-22 10:01:27.000000000 +0100
+++ b/mt7603_wifi_ap/Makefile	2019-12-08 21:52:10.253012796 +0100
@@ -16,6 +16,90 @@ DRV_NAME = mt7603_wifi
 RT_WIFI_DIR = ../mt7603_wifi
 endif
 
+ifeq ($(CONFIG_SUPPORT_OPENWRT),y)
+###EXTRA_CFLAGS += -DCONFIG_SUPPORT_OPENWRT
+ifneq ($(CONFIG_RT_FIRST_CARD),)
+EXTRA_CFLAGS += -DCONFIG_RT_FIRST_CARD=$(CONFIG_RT_FIRST_CARD)
+EXTRA_CFLAGS += -DCONFIG_RT_FIRST_IF_RF_OFFSET=$(CONFIG_RT_FIRST_IF_RF_OFFSET)
+endif
+ifneq ($(CONFIG_RT_SECOND_CARD),)
+EXTRA_CFLAGS += -DCONFIG_RT_SECOND_CARD=$(CONFIG_RT_SECOND_CARD)
+EXTRA_CFLAGS += -DCONFIG_RT_SECOND_IF_RF_OFFSET=$(CONFIG_RT_SECOND_IF_RF_OFFSET)
+endif
+ifneq ($(CONFIG_RT_THIRD_CARD),)
+EXTRA_CFLAGS += -DCONFIG_RT_THIRD_CARD=$(CONFIG_RT_THIRD_CARD)
+EXTRA_CFLAGS += -DCONFIG_RT_THIRD_IF_RF_OFFSET=$(CONFIG_RT_THIRD_IF_RF_OFFSET)
+endif
+ifeq ($(CONFIG_FIRST_IF_EEPROM_PROM),y)
+EXTRA_CFLAGS += -DCONFIG_RT_FIRST_CARD_EEPROM="\"prom\""
+else
+ifeq ($(CONFIG_FIRST_IF_EEPROM_EFUSE),y)
+EXTRA_CFLAGS += -DCONFIG_RT_FIRST_CARD_EEPROM="\"efuse\""
+else
+ifeq ($(CONFIG_FIRST_IF_EEPROM_FLASH),y)
+EXTRA_CFLAGS += -DCONFIG_RT_FIRST_CARD_EEPROM="\"flash\""
+endif
+endif
+endif
+ifeq ($(CONFIG_SECOND_IF_EEPROM_PROM),y)
+EXTRA_CFLAGS += -DCONFIG_RT_SECOND_CARD_EEPROM="\"prom\""
+else
+ifeq ($(CONFIG_SECOND_IF_EEPROM_EFUSE),y)
+EXTRA_CFLAGS += -DCONFIG_RT_SECOND_CARD_EEPROM="\"efuse\""
+else
+ifeq ($(CONFIG_SECOND_IF_EEPROM_FLASH),y)
+EXTRA_CFLAGS += -DCONFIG_RT_SECOND_CARD_EEPROM="\"flash\""
+endif
+endif
+endif
+ifeq ($(CONFIG_THIRD_IF_EEPROM_PROM),y)
+EXTRA_CFLAGS += -DCONFIG_RT_THIRD_CARD_EEPROM="\"prom\""
+else
+ifeq ($(CONFIG_THIRD_IF_EEPROM_EFUSE),y)
+EXTRA_CFLAGS += -DCONFIG_RT_THIRD_CARD_EEPROM="\"efuse\""
+else
+ifeq ($(CONFIG_THIRD_IF_EEPROM_FLASH),y)
+EXTRA_CFLAGS += -DCONFIG_RT_THIRD_CARD_EEPROM="\"flash\""
+endif
+endif
+endif
+endif
+
+ifeq ($(CONFIG_FIRST_IF_EPAELNA),y)
+EXTRA_CFLAGS += -DCONFIG_FIRST_IF_EPAELNA
+else
+ifeq ($(CONFIG_FIRST_IF_IPAILNA),y)
+EXTRA_CFLAGS += -DCONFIG_FIRST_IF_IPAILNA
+else
+ifeq ($(CONFIG_FIRST_IF_IPAELNA),y)
+EXTRA_CFLAGS += -DCONFIG_FIRST_IF_IPAELNA
+endif
+endif
+endif
+ifeq ($(CONFIG_SECOND_IF_EPAELNA),y)
+EXTRA_CFLAGS += -DCONFIG_SECOND_IF_EPAELNA
+else
+ifeq ($(CONFIG_SECOND_IF_IPAILNA),y)
+EXTRA_CFLAGS += -DCONFIG_SECOND_IF_IPAILNA
+else
+ifeq ($(CONFIG_SECOND_IF_IPAELNA),y)
+EXTRA_CFLAGS += -DCONFIG_SECOND_IF_IPAELNA
+endif
+endif
+endif
+ifeq ($(CONFIG_THIRD_IF_EPAELNA),y)
+EXTRA_CFLAGS += -DCONFIG_THIRD_IF_EPAELNA
+else
+ifeq ($(CONFIG_THIRD_IF_IPAILNA),y)
+EXTRA_CFLAGS += -DCONFIG_THIRD_IF_IPAILNA
+else
+ifeq ($(CONFIG_THIRD_IF_IPAELNA),y)
+EXTRA_CFLAGS += -DCONFIG_THIRD_IF_IPAELNA
+endif
+endif
+endif
+
+
 ########################################################
 # Common files
 ########################################################
