From 58f33070591a5a11b532b8365874d0eb104b0aab Mon Sep 17 00:00:00 2001
From: Anjan Chanda <anjan.chanda@iopsys.eu>
Date: Sun, 23 Feb 2020 12:30:45 +0100
Subject: [PATCH] disable multi profile and set default to 5G

---
 mt_wifi_ap/Makefile | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/mt_wifi_ap/Makefile b/mt_wifi_ap/Makefile
index a49520d..f8f16ab 100644
--- a/mt_wifi_ap/Makefile
+++ b/mt_wifi_ap/Makefile
@@ -192,12 +192,12 @@ cmm_objs := $(SRC_EMBEDDED_DIR)/common/action.o\
 		$(SRC_DIR)/protocol/protection.o\
 		$(SRC_EMBEDDED_DIR)/common/misc_app.o
 
-EXTRA_CFLAGS += -DMULTI_PROFILE
-cmm_objs += $(SRC_EMBEDDED_DIR)/common/multi_profile.o
+#EXTRA_CFLAGS += -DMULTI_PROFILE
+#cmm_objs += $(SRC_EMBEDDED_DIR)/common/multi_profile.o
 
-ifeq ($(CONFIG_DEFAULT_5G_PROFILE),y)
+#ifeq ($(CONFIG_DEFAULT_5G_PROFILE),y)
 EXTRA_CFLAGS += -DDEFAULT_5G_PROFILE
-endif
+#endif
 
 ########################################################
 # Coex related files
-- 
2.7.4

