diff -Nur picoLisp/Makefile picoLisp-new/Makefile
--- picoLisp/Makefile	1969-12-31 21:00:00.000000000 -0300
+++ picoLisp-new/Makefile	2015-06-02 15:38:07.469726899 -0300
@@ -0,0 +1,6 @@
+all:
+	(cd src; make)
+
+clean:
+	(cd src; make clean)
+
diff -Nur picoLisp/src/Makefile picoLisp-new/src/Makefile
--- picoLisp/src/Makefile	2015-03-31 15:48:22.000000000 -0300
+++ picoLisp-new/src/Makefile	2015-06-02 15:40:43.541491599 -0300
@@ -6,9 +6,13 @@
 
 picoFiles = main.c gc.c apply.c flow.c sym.c subr.c big.c io.c net.c tab.c
 
-CC = gcc
+#CC = gcc
 # CCLD is the cc (compiler frontend) to use for the link step.
-CCLD = gcc
+#CCLD = gcc
+
+# Honor the $(CC) variable, inherited form the environment (it's essential
+# when cross-compiling)
+CCLD = $(CC)
 
 CFLAGS = -c -O2 -pipe \
 	-falign-functions=32 -fomit-frame-pointer -fno-strict-aliasing \
@@ -19,10 +23,10 @@
 
 ifeq ($(shell uname), Linux)
 	OS = Linux
-	CFLAGS += -m32
-	PICOLISP-FLAGS = -m32 -rdynamic
+	CFLAGS += -mabi=32 -fPIC
+	PICOLISP-FLAGS = -rdynamic
 	LIB-FLAGS = -lm -ldl
-	DYNAMIC-LIB-FLAGS = -m32 -shared -export-dynamic
+	DYNAMIC-LIB-FLAGS = -shared -export-dynamic -fPIC
 	LCRYPT = -lcrypt
 	STRIP = strip
 else
